added llvm3.9 packages to installation in case of fedora 27

speedskater opened the pull request:
I had problems building peloton with fedora 27 due to fedora's default LLVM Version (could be related to issue #1106 ). Therefore I propose this changes to the package.sh install script to automatically install LLVM3.9 on fedora 27.

Additionally I would like you to ask, if it would be possible to add the following Line to the wiki for building peloton with Fedora (27):

cmake -DLLVM_DIR=/usr/lib64/cmake/llvm3.9/ -DCMAKE_BUILD_TYPE=Release ../

With this two changes building peloton works on my fedora 27 system out of the box.

crd477 commented:
The Fedora 27 Docker image in our Jenkins setup was already configured with LLVM 4.0 before they updated it to LLVM 5.0 so I suggest [a slightly different patch](https://github.com/cmu-db/peloton/files/1690756/fedora-llvm.diff.txt).

speedskater commented:
@crd477 I tried to incorporate your suggestion, but unfortuntely there seems to be a problem with the cmake configuration provided with the llvm4.0 package in fedora 27 resulting in the following build error:
```/home/rbinna/VCS/peloton/src/include/codegen/code_context.h:18:10: fatal error: llvm/IR/IRBuilder.h: No such file or directory
 #include "llvm/IR/IRBuilder.h"
```
Therefore, I changed the script to use LLVM 3.9 for all fedora versions >= 27. Furthermore I updated the fedora 27 docker file to the use the LLVM 3.9 version to build peloton. 

Do you have any other idea to build peloton with llvm >= 4.0 ?

pervazea commented:
We have decided that we are not going to support Fedora at this time.
